tokens/usecase/tokens_usecase.go

```diff
@@ -302,41 +352,11 @@ func GetTokensFromChainRegistry(chainRegistryAssetsFileURL string) (map[string]d
 	// Iterate through each asset and its denom units to print exponents
 	for _, asset := range assetList.Assets {
 		token := domain.Token{}
-		chainDenom := ""
-
-		for _, denom := range asset.DenomUnits {
-			if denom.Exponent == 0 {
-				chainDenom = denom.Denom
-
-				if len(asset.DenomUnits) == 1 {
-					token.Precision = denom.Exponent
-				}
-			}
-
-			if denom.Exponent > 0 {
-				// There are edge cases where we have 3 denom exponents for a token.
-				// We filter out the intermediate denom exponents and only use the human readable denom.
-				if denom.Denom == "mluna" || denom.Denom == "musd" || denom.Denom == "msomm" || denom.Denom == "mkrw" || denom.Denom == "uarch" {
-					continue
-				}
-
-				token.Precision = denom.Exponent
-			}
-		}
-
-		// Detect unlisted tokens
-		isUnlisted := false
-		for _, keyword := range asset.Keywords {
-			if keyword == unlistedKeyword {
-				isUnlisted = true
-				break
-			}
-		}
-
+		token.Precision = asset.Decimals
 		token.HumanDenom = asset.Symbol
-		token.IsUnlisted = isUnlisted
-
-		tokensByChainDenom[chainDenom] = token
+		token.IsUnlisted = asset.Disabled
+		token.CoingeckoID = asset.CoingeckoID
+		tokensByChainDenom[asset.CoinMinimalDenom] = token
```

Comment:
Looks like v2 has a much cleaner format

Reply:
Please note ``is_unlisted`` maps to ``preview``, not ``disabled``. Fixed.